### PR TITLE
[RLlib] Make rollout always use `evaluation_config`.

### PR DESCRIPTION
**Pull request by sven1977**
Make rollout always use evaluation config. Set `in_evaluation=True` in default evaluation_config.

Clarifies question https://github.com/ray-project/ray/issues/7394

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: rllib/agents/trainer.py [hunk 1]
```diff
@@ -197,6 +197,8 @@
     # policy, even if this is a stochastic one. Setting "explore=False" here
     # will result in the evaluation workers not using this optimal policy!
     "evaluation_config": {
+        # Set `in_evaluation` flag to True, in case a Trainer needs to know.
+        "in_evaluation": True,
```
Comment (ericl) [inline rllib/agents/trainer.py]: Add a comment that this should not be set by the user.

File: rllib/agents/trainer.py [hunk 2]
```diff
@@ -594,7 +596,6 @@ def get_scope():
                 extra_config.update({
                     "batch_mode": "complete_episodes",
                     "batch_steps": 1,
-                    "in_evaluation": True,
```
Comment (ericl) [inline rllib/agents/trainer.py]: Can we still force set this to avoid user error (or at least assert).

File: rllib/agents/trainer.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/trainer.py]: Ok. Done.

File: rllib/agents/trainer.py (hunk 2 above)
Comment (sven1977) [reply rllib/agents/trainer.py]: Sure. Good idea.

**Review by ericl — APPROVED** [review]

**Pull request merged**